Source/Scene/Scene.js

```diff
@@ -3504,20 +3523,34 @@ define([
             limit = Number.MAX_VALUE;
         }
 
-        var pickedResult = this.pick(windowPosition, width, height);
-        while (defined(pickedResult) && defined(pickedResult.primitive)) {
-            result.push(pickedResult);
-            if (0 >= --limit) {
+        var pickedResult = pickCallback();
+        while (defined(pickedResult)) {
+            var object = pickedResult.object;
+            var position = pickedResult.position;
```

Comment:
Instead of the callback returning an anonymous object, wouldn't it be better to do `if (result instanceof Cartesian3) { // do something with position}`

It doesn't look like both position and object are ever used, it seems to be one or the other

Reply:
Oh nevermind, I guess this is the object sent back to some of your other functions too